java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -1784,7 +1784,11 @@ public void didOpen(DidOpenTextDocumentParams params) {
                 client.logMessage(new MessageParams(MessageType.Error, ex.getMessage()));
             }
             openedDocuments.put(params.getTextDocument().getUri(), doc);
-            runDiagnoticTasks(params.getTextDocument().getUri());
+            
+            // attempt to open the directly owning project, delay diagnostics after project open:
```

Comment:
Possibly this could open a completely unrelated project - will the user be asked for a consent to trust the project in that case?

Btw. there is [NETBEANS-5204](https://github.com/apache/netbeans/pull/2629/) to help you find whether a subproject is related to already opened project or not.

Reply:
Good point with the project relationships; will try to add checks.